nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/BatchUploadObject.java

```diff
@@ -292,10 +307,14 @@ public Object execute(@PathParam(REQUEST_BATCH_ID) String batchId, @PathParam(RE
 
     protected Object executeBatch(String batchId, String fileIdx, String operationId, HttpServletRequest request,
             ExecutionRequest xreq) throws UnsupportedEncodingException {
-        RequestContext.getActiveContext(request).addRequestCleanupHandler(req -> {
-            BatchManager bm = Framework.getService(BatchManager.class);
-            bm.clean(batchId);
-        });
+
+        if (!Boolean.parseBoolean(
+            RequestContext.getActiveContext(request).getRequest().getHeader(BatchManagerConstants.NO_DROP_FLAG))) {
+            RequestContext.getActiveContext(request).addRequestCleanupHandler(req -> {
+                BatchManager bm = Framework.getService(BatchManager.class);
+                bm.clean(batchId);
+            });
+        }
```

Comment:
The code is ok, but in case the files are not removed as part of the batch and if the "client" does not properly call the DELETE endpoint: who will do the cleanup ?